### DISCUSSION TITLE
Trimming / Delimiting values based on character

### DISCUSSION THREAD

**Discussion by sr320** [RobertsLab/resources#1452] [Q&A] [answered]
I have a column with in the following format

```
lcl|NC_047561.1_prot_XP_034327014.1_14848
lcl|NC_047568.1_prot_XP_034310977.1_52469
lcl|NC_047565.1_prot_XP_034301456.1_43412
```
but want 

```
lcl|NC_047561.1_prot_XP_034327014
lcl|NC_047568.1_prot_XP_034310977
lcl|NC_047565.1_prot_XP_034301456
```

ie removing everything after the last `.`.

How can I do this in tidyverse?

**Comment by kubu4** [comment]
```R
# creates three column data frame
df <- read.csv(file = "your-file.txt", sep = ".")

# adds new column with tailing junk removed
df$concatenated <- paste(df$V1, df$V2, sep = ".")
```

**Reply by sr320** [reply]
Thanks!
But original table has several columns so reading in like that would be a mess- though could make work.

Wondering if something similar could be done post reading in.

**Reply by kubu4** [reply]
Can you post example of full file?

**Reply by sr320** [reply]
just figured out `separate` will work 🤦‍♂️ - but here is full file https://raw.githubusercontent.com/sr320/nb-2022/main/P_generosa/analyses/gene-duck-blastx-gigas.tab

**Comment by sr320** [comment, marked as answer]
🤦‍♂️ I teach this..
`separate` and `unite` will do the trick. will post code here once raven is back

**Reply by kubu4** [reply]
Great!

But, what's wrong with Raven??

**Reply by sr320** [reply]
could be more elegant but `gigloc %>% separate(V2, sep = "[.]", into = c("a","b","c")) %>% unite(new, a, b,sep = "." )`

>But, what's wrong with Raven?? 

I was running a long chunk thus could not test
